### PR TITLE
Basic Refactoring

### PR DESCRIPTION
**Pull request by elyesmanai**
- added utils.py
- put the independant functions there
- did some code organisation

code is tested, successfully scrapped data.

### CODE REVIEW

**Comment by harismuneer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- scraper/utils.py  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/harismuneer/Ultimate-Facebook-Scraper/pullRequest?prid=5254804&bid=17218068)

File: scraper/utils.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+            if link != "None":
+                img_name = (link.split(".jpg")[0]).split("/")[-1] + ".jpg"
+
+                # this is the image id when there's no profile pic
+                if img_name == selectors.get("default_image"):
+                    img_name = "None"
+                else:
+                    try:
+                        urllib.request.urlretrieve(link, img_name)
```
Comment (harismuneer) [inline scraper/utils.py]: Codacy Issue found: [Audit url open for permitted schemes. Allowing use of file:/ or custom schemes is often unexpected.](https://app.codacy.com/manual/harismuneer/Ultimate-Facebook-Scraper/pullRequest?prid=5254804)

File: scraper/scraper.py [hunk 2]
```diff
@@ -82,86 +99,236 @@ def get_facebook_images_url(img_links):
...
+            f.writelines(results)
 
+        # dealing with Posts
+        elif status == 4:
+            extract_and_write_posts(elements, name)
+            return
 
-# -------------------------------------------------------------
-# -------------------------------------------------------------
+        """Write results to file"""
```
Comment (harismuneer) [inline scraper/scraper.py]: Codacy Issue found: [String statement has no effect](https://app.codacy.com/manual/harismuneer/Ultimate-Facebook-Scraper/pullRequest?prid=5254804)

**Comment by harismuneer** [comment]
Hi @elyesmanai 

Thanks for the PR! Can you please give these issues a look and fix them too. They are pretty minor ones but crucial. Just comment below when done and I will merge the PR. 

https://github.com/harismuneer/Ultimate-Facebook-Scraper/issues/92
https://github.com/harismuneer/Ultimate-Facebook-Scraper/issues/91

**Comment by harismuneer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- scraper/utils.py  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/harismuneer/Ultimate-Facebook-Scraper/pullRequest?prid=5254804&bid=17218068)

File: scraper/scraper.py [hunk 3]
```diff
@@ -869,17 +597,104 @@ def login(email, password):
...
+    if save_status == 4:
+        page.append(user_id)
+
+    page += [user_id + s for s in section]
+
+    for i, _ in enumerate(scan_list):
+        try:
+            driver.get(page[i])
+            assert "facebook" in driver.title
+
```
Comment (harismuneer) [inline scraper/scraper.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/harismuneer/Ultimate-Facebook-Scraper/pullRequest?prid=5254804)

File: scraper/utils.py (hunk 1 above)
Comment (harismuneer) [inline scraper/utils.py]: Codacy Issue found: [Audit url open for permitted schemes. Allowing use of file:/ or custom schemes is often unexpected.](https://app.codacy.com/manual/harismuneer/Ultimate-Facebook-Scraper/pullRequest?prid=5254804)

File: scraper/scraper.py [hunk 4]
```diff
@@ -82,86 +102,236 @@ def get_facebook_images_url(img_links):
...
+            f.writelines(results)
 
+        # dealing with Posts
+        elif status == 4:
+            extract_and_write_posts(elements, name)
+            return
 
-# -------------------------------------------------------------
-# -------------------------------------------------------------
+        """Write results to file"""
```
Comment (harismuneer) [inline scraper/scraper.py]: Codacy Issue found: [String statement has no effect](https://app.codacy.com/manual/harismuneer/Ultimate-Facebook-Scraper/pullRequest?prid=5254804)

File: scraper/scraper.py [hunk 5]
```diff
@@ -869,17 +597,104 @@ def login(email, password):
...
+
+    if save_status == 4:
+        page.append(user_id)
+
+    page += [user_id + s for s in section]
+
+    for i, _ in enumerate(scan_list):
+        try:
+            driver.get(page[i])
+            assert "facebook" in driver.title
```
Comment (harismuneer) [inline scraper/scraper.py]: Codacy Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/harismuneer/Ultimate-Facebook-Scraper/pullRequest?prid=5254804)

**Comment by harismuneer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- scraper/utils.py  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/harismuneer/Ultimate-Facebook-Scraper/pullRequest?prid=5254804&bid=17218068)

File: scraper/utils.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
```
Comment (harismuneer) [inline scraper/utils.py]: Codacy Issue found: [Audit url open for permitted schemes. Allowing use of file:/ or custom schemes is often unexpected.](https://app.codacy.com/manual/harismuneer/Ultimate-Facebook-Scraper/pullRequest?prid=5254804)

**Comment by elyesmanai** [comment]
okay so following #91, It doesn't work for me at all, not one value is scrapped.
By following the trace I found it was the driver.get(page[i]) in scrap_data() that's returning none.
I don't know if I'm the only one and anyways I currently lack the knowledge to fix this & time to learn this so if you could please see this yourself as you know your code better.

**Comment by harismuneer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- scraper/utils.py  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/harismuneer/Ultimate-Facebook-Scraper/pullRequest?prid=5254804&bid=17218068)

File: scraper/utils.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+            if link != "None":
+                img_name = (link.split(".jpg")[0]).split("/")[-1] + ".jpg"
+
+                # this is the image id when there's no profile pic
+                if img_name == selectors.get("default_image"):
+                    img_name = "None"
+                else:
+                    try:
+                        req = urllib.request.Request(link)
+                        with urllib.request.urlopen(req) as response:
```
Comment (harismuneer) [inline scraper/utils.py]: Codacy Issue found: [Audit url open for permitted schemes. Allowing use of file:/ or custom schemes is often unexpected.](https://app.codacy.com/manual/harismuneer/Ultimate-Facebook-Scraper/pullRequest?prid=5254804)

**Comment by harismuneer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- scraper/utils.py  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/harismuneer/Ultimate-Facebook-Scraper/pullRequest?prid=5254804&bid=17218068)

File: scraper/utils.py (hunk 6 above)
Comment (harismuneer) [inline scraper/utils.py]: Codacy Issue found: [Audit url open for permitted schemes. Allowing use of file:/ or custom schemes is often unexpected.](https://app.codacy.com/manual/harismuneer/Ultimate-Facebook-Scraper/pullRequest?prid=5254804)

**Comment by harismuneer** [comment]
@elyesmanai I will look in #91 myself when I get some time. 

Just ping me on LinkedIn when you are done, I appreciate the refactoring you did :)

**Pull request closed without merging**